include/deal.II/lac/trilinos_tpetra_precondition.h

```diff
@@ -1316,6 +1320,176 @@ namespace LinearAlgebra
+         * @brief The overlap between the subdomains.
+         *
+         */
+        int overlap;
+        /**
+         * @brief The dimension of the problem
+         *
+         */
+        int dimension;
+        /**
+         * @brief The combine method in the overlap.
+         *
+         * Tell FROSch how to combine the values from the overlap.
+         * When using "Restricted" as combine mode, the resulting
+         * preconditioner will not be symmetrical, and therefore can not
+         * be used along with CG and rather has to be used with GMRES.
+         * However, Overlapping Restricted Additive Schwarz (ORAS)
+         * preconditioners lead to better condition numbers.
+         *
+         * So a good choice for the combine mode for non-symmetical problems is
+         * "Restricted". To preserve the symmetry the combine Method "Full" can
+         * be used.
+         *
+         * The options are:
+         * <ul>
+         * <li> "Restricted" </li>
+         * <li> "Averaging" </li>
+         * <li> "Full" </li>
+         * </ul>
+         */
+        std::string combine_values_in_overlap;
+
+        /**
+         * @brief Specify the direct solver for the subdomains.
+         *
+         * The available options depend on how Trilinos was configured.
+         * Some common available options are:
+         * <ul>
+         * <li> "KLU" (default) </li>
+         * <li> "UMFPACK" </li>
+         * <li> "MUMPS" (requires Trilinos Amesos2 to be configured to use MUMPS) </li>
+         * <li> "SuperLU_dist" (requires Trilinos Amesos2 to be configured to use SuperLU_dist) </li>
+         * </ul>
+         *
+         */
+        std::string subdomain_solver;
+
+        /**
+         * @brief Specify the coarse space operator. Only used by two-level methods.
+         *
+         * Available options:
+         * <ul>
+         * <li> "IPOUHarmonicCoarseOperator" </li>
+         * <li> "GDSWCoarseOperator" (Generalized Dryja–Smith–Widlund) </li>
+         * <li> "RGDSWCoarseOperator" (Reduced dimension GDSW) </li>
+         * </ul>
+         *
+         */
+        std::string coarse_operator_type;
+
+        /**
+         * @brief The solver used for the extension space problem.
+         * For the available options, see the description of the
+         * subdomain_solver.
+         */
+        std::string extension_solver;
+
+        /**
+         * @brief The solver used for the coarse space problem.
+         * For the available options, see the description of the
+         * subdomain_solver.
+         */
+        std::string coarse_solver;
+      };
+
+      /**
+       * @brief Construct a FROSch preconditioner.
+       *
+       *  Available options:
+       *  <ul>
+       *  <li> "One Level" </li>
+       *  <li> "Two Level" </li>
+       *  </ul>
+       *
+       * @param precondition_type the type of FROSch preconditioner to use
+       */
+      PreconditionFROSch(std::string precondition_type = "One Level");
```

Comment:
Same thing here about the string. Separately, mark all input argument as `const` and, if the type of the object is large, as a reference.